modules/c++/re/source/RegexPCRE.cpp

```diff
@@ -70,244 +181,145 @@ re::Regex& re::Regex::operator=(const re::Regex& rhs)
-                           OVECTOR_COUNT);                    // output vector size
-
-    result = numMatches;
-
-    if (result == 0)
-    {
-        numMatches = OVECTOR_COUNT / 3;
-    }
-
-    if (result >= 0)
-    {
-        if (((mOvector[0] + startIndex) +
-             (mOvector[1] - mOvector[0])) >
-            (int)matchString.length() )
-        {
-            result = PCRE_ERROR_NOMATCH;
-        }
-    }
+    ScopedMatchData matchData(mPCRE);
+    const size_t numMatches = matchData.match(matchString, startIndex, flags);
 
-    if (result >= 0)
-    {
-        // output vector start offset = i*2
-        // output vector end   offset = i*2+1
-        // i = 0
-        int index = mOvector[0] + startIndex;
-        int subStringLength = mOvector[1] - mOvector[0];
-        int subStringCheck = index + subStringLength;
-        if (subStringCheck > (int)matchString.length())
-        {
-            throw RegexException(Ctxt(FmtX("Search: Match substring out of range (%d,%d) for string of length %d", index, subStringCheck, matchString.length())));
-        }
-        return matchString.substr(index, subStringLength);
-    }
-    else if (result == PCRE_ERROR_NOMATCH)
+    if (numMatches > 0)
     {
-        return std::string("");
+        begin = matchData.getOutputVector()[0];
+        end = matchData.getOutputVector()[1];
+        return matchData.getMatch(matchString, 0);
     }
     else
     {
-        throw RegexException
-            (Ctxt(FmtX("Error in searching for %s", matchString.c_str())));
+        begin = end = 0;
+        return "";
     }
 }
 
-void re::Regex::searchAll(const std::string& matchString,
-                          RegexMatch & v)
+void Regex::searchAll(const std::string& matchString, RegexMatch& v)
 {
-    std::string result = search(matchString, 0, 0);
+    size_t startIndex = 0;
+
+    size_t begin;
+    size_t end;
+    std::string result = search(matchString, startIndex, 0, begin, end);
 
-    int idx = 0;
-    while (result.size() != 0)
+    while (!result.empty())
     {
         v.push_back(result);
-        idx += (int)result.size() + mOvector[0];
-        result = search(matchString, idx, PCRE_NOTBOL);
+        startIndex = end;
+        result = search(matchString, startIndex, PCRE2_NOTBOL, begin, end);
     }
 }
 
-void re::Regex::split(const std::string& str,
-                      std::vector<std::string> & v)
+void Regex::split(const std::string& str, std::vector<std::string>& v)
 {
-    size_t idx = 0;
-    std::string result = search(str, 0, 0);
-    while (result.size() != 0)
+    size_t begin;
+    size_t end;
+    size_t startIndex = 0;
+    std::string result = search(str, startIndex, 0, begin, end);
+    while (!result.empty())
     {
-        v.push_back(str.substr(idx, mOvector[0]));
-        idx += mOvector[1];
-        result = search(str, idx, PCRE_NOTBOL);
+        v.push_back(str.substr(startIndex, end - begin + 1));
+        startIndex = end;
```

Comment:
Same deal with what `end` means, but based on the note below, something was incorrect in the indexing here anyhow... should be fixed now